packages/caver-transaction/src/transactionTypes/abstractFeeDelegatedTransactionType.js

```diff
@@ -0,0 +1,222 @@
+/*
+    Copyright 2020 The caver-js Authors
+    This file is part of the caver-js library.
+
+    The caver-js library is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    The caver-js library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with the caver-js. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+const _ = require('lodash')
+const RLP = require('eth-lib/lib/rlp')
+const Hash = require('eth-lib/lib/hash')
+const Bytes = require('eth-lib/lib/bytes')
+const TransactionHasher = require('../transactionHasher/transactionHasher')
+const AbstractTransactionType = require('./abstractTransactionType')
+const { refineSignatures, typeDetectionFromRLPEncoding } = require('../transactionHelper/transactionHelper')
+const formatters = require('../../../caver-core-helpers/src/formatters')
+const Keyring = require('../../../caver-wallet/src/keyring/keyring')
+const { KEY_ROLE } = require('../../../caver-wallet/src/keyring/keyringHelper')
+const utils = require('../../../caver-utils')
+
+/**
+ * Abstract class that implements common logic for each fee delegated transaction type.
+ * @class
+ */
+class AbstractFeeDelegatedTransactionType extends AbstractTransactionType {
```

Comment:
type suffix is necessary? What if we remove "Type" at the end?

Reply:
i changed name! :)